src/imports/register-extensions.js

```diff
@@ -15,36 +15,54 @@ import logger from './logger';
  * @param {string} keyword - Keyword to use when searching for extensions to
  * register.
  * @param {Object} object - Object to pass to the register function.
- * @param {boolean} [includeGlobal=false] - Determines if global packages
+ * @param {Object} options - Object with optional paramsts
+ * @param {boolean} options.includeGlobal - Determines if global packages
  * should be registered.
- * @param {string} [registerFunction=register] - Name of the register function
+ * @param {string} options.registerFunction - Name of the register function
  * used during extension registration.
  */
-const registerExtensions = (keyword, object, includeGlobal = false,
-  registerFunction = 'register') => {
+const registerExtensions = (keyword, object, options = {}) => {
+  const includeGlobal = options.includeGlobal || false;
+  const registerFunction = options.registerFunction || 'register';
+
   const dependencyNames = [
     ...Object.keys(pkgConf.sync('dependencies')),
     ...Object.keys(pkgConf.sync('devDependencies')),
   ];
 
   const dependencies = dependencyNames.map(name => {
+    let dependencyPath;
+
+    const resolveOptions = {};
+
+    if (object.projectRoot) {
+      resolveOptions.basedir = object.projectRoot;
+    }
+
     try {
-      return resolve.sync(name);
+      dependencyPath = resolve.sync(name, resolveOptions);
     } catch (error) {
-      logger.info(`Error trying to resolve dependency ${name}. Continuing...`);
+      logger.debug(`Error trying to resolve dependency ${name}. Continuing...`);
```

Comment:
Typically considered bad practice to swallow the error here.  We are assuming that error information is not important for debugging, but giving the error might help someone during development figure out what they did incorrectly or even someone working on the project in the future.  We could provide the error when only in debug mode if we are worried about cluttering log output.